File: src/csharp/pulse/cdm/system/equipment/anesthesia_machine/actions/SEAnesthesiaMachineConfiguration.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SEAnesthesiaMachineConfiguration : SEAnesthesiaMachineAction
  {
    protected SEAnesthesiaMachine configuration = null;
    protected string configurationFile = "";

    public SEAnesthesiaMachineConfiguration()
    {

    }

    public SEAnesthesiaMachineConfiguration(SEAnesthesiaMachineConfiguration other)
    {
      Copy(other);
    }

    public void copy(SEAnesthesiaMachineConfiguration other)
    {
      base.Copy(other);
      if(other.configuration != null)
        this.GetConfiguration().Copy(other.configuration);
      this.configurationFile = other.configurationFile;
    }

    public override void Clear()
    {
      base.Clear();
      if (this.configuration != null)
        this.configuration.Clear();
      this.configurationFile = "";
    }

    public override bool IsValid()
    {
      return HasConfiguration() || HasConfigurationFile();
    }


    public bool HasConfiguration()
    {
      return this.configuration != null;
    }
    public SEAnesthesiaMachine GetConfiguration()
    {
      if (this.configuration == null)
        this.configuration = new SEAnesthesiaMachine();
      return this.configuration;
    }

    public bool HasConfigurationFile()
    {
      return !string.IsNullOrEmpty(this.configurationFile);
    }
    public string GetConfigurationFile()
    {
      return this.configurationFile;
    }
    public void SetConfigurationFile(string s)
    {
      this.configurationFile = s;
    }

    public override string ToString()
    {
      string str = "Anesthesia Machine Configuration";
      if (this.HasConfigurationFile())
        str += "\n\tAnesthesia Machine File: " + this.configurationFile;
      else if (HasConfiguration())
      {
        str += configuration.ToString();
      }
      
      return str;
    }
  }
}

File: src/java/com/kitware/pulse/cdm/system/equipment/anesthesia_machine/actions/SEAnesthesiaMachineConfiguration.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.system.equipment.anesthesia_machine.actions;
import com.kitware.pulse.cdm.bind.AnesthesiaMachineActions.AnesthesiaMachineConfigurationData;
import com.kitware.pulse.cdm.system.equipment.anesthesia_machine.SEAnesthesiaMachine;
import com.kitware.pulse.utilities.Log;

public class SEAnesthesiaMachineConfiguration extends SEAnesthesiaMachineAction
{

  private static final long serialVersionUID = 7917788116628472277L;
  protected SEAnesthesiaMachine configuration=null;
  protected String              configurationFile="";
  
  public SEAnesthesiaMachineConfiguration()
  {
    
  }
  
  public SEAnesthesiaMachineConfiguration(SEAnesthesiaMachineConfiguration other)
  {
    this();
    copy(other);
  }
  
  public void copy(SEAnesthesiaMachineConfiguration other)
  {
    if(this==other)
      return;
    super.copy(other);
    if(other.configuration!=null)
      this.getConfiguration().copy(other.configuration);
    this.configurationFile=other.configurationFile;
  }
  
  @Override
  public void reset()
  {
    super.reset();
    
    if (this.configuration != null)
      this.configuration.reset();
    this.configurationFile="";
  }
  
  @Override
  public boolean isValid()
  {
    return hasConfiguration() || hasConfigurationFile();
  }
  
  public static void load(AnesthesiaMachineConfigurationData src, SEAnesthesiaMachineConfiguration dst)
  {
    SEAnesthesiaMachineAction.load(src.getAnesthesiaMachineAction(),dst);
    switch(src.getOptionCase())
    {
    case CONFIGURATIONFILE:
      dst.configurationFile = src.getConfigurationFile();
      break;
    case CONFIGURATION:
      SEAnesthesiaMachine.load(src.getConfiguration(),dst.getConfiguration());
      break;
    default:
    	Log.error("Unknown AnesthesiaMachineConfigurationData Option");
    }
  }
  public static AnesthesiaMachineConfigurationData unload(SEAnesthesiaMachineConfiguration src)
  {
    AnesthesiaMachineConfigurationData.Builder dst = AnesthesiaMachineConfigurationData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEAnesthesiaMachineConfiguration src, AnesthesiaMachineConfigurationData.Builder dst)
  {
    SEAnesthesiaMachineAction.unload(src, dst.getAnesthesiaMachineActionBuilder());
    if(src.hasConfigurationFile())
      dst.setConfigurationFile(src.configurationFile);
    else if(src.hasConfiguration())
      dst.setConfiguration(SEAnesthesiaMachine.unload(src.configuration));
  }
  
  public boolean hasConfiguration()
  {
    return this.configuration!=null;
  }
  public SEAnesthesiaMachine getConfiguration()
  {
    if(this.configuration==null)
      this.configuration=new SEAnesthesiaMachine();
    return this.configuration;
  }
  
  public boolean hasConfigurationFile()
  {
    return this.configurationFile!=null&&!this.configurationFile.isEmpty();
  }
  public String getConfigurationFile()
  {
    return this.configurationFile;
  }
  public void setConfigurationFile(String s)
  {
    this.configurationFile = s;
  }
  
  @Override
  public String toString()
  {
    String str = "Anesthesia Machine Configuration";
    if(this.hasConfigurationFile())
      str +="\n\tAnesthesia Machine File: "+this.configurationFile;
    else if(hasConfiguration())
    {
      str += configuration.toString();
    }
    
    return str;
  }
}